src/main/java/io/kestra/plugin/pulsar/Produce.java

```diff
@@ -146,156 +130,43 @@ public class Produce extends AbstractPulsarConnection implements RunnableTask<Pr
-        }
-    }
-
-    @SuppressWarnings("unchecked")
-    private Flowable<Integer> buildFlowable(Flowable<Object> flowable, RunContext runContext, Producer<byte[]> producer) {
-        return flowable
-            .map(row -> {
-                this.produceMessage(producer, runContext, (Map<String, Object>) row);
-                return 1;
-            });
-    }
-
-    @SuppressWarnings("unchecked")
-    private CompletableFuture<MessageId> produceMessage(Producer<byte[]> producer, RunContext runContext, Map<String, Object> map) throws Exception {
-        TypedMessageBuilder<byte[]> message = producer.newMessage();
-        Map<String, Object> renderedMap = runContext.render(map);
-
-        if (renderedMap.containsKey("key")) {
-            message.key((String) renderedMap.get("key"));
-        }
-
-        if (renderedMap.containsKey("properties")) {
-            message.properties((Map<String, String>) renderedMap.get("properties"));
-        }
-
-        if (renderedMap.containsKey("value")) {
-            message.value(this.serializer.serialize(renderedMap.get("value")));
-        }
-
-        if (renderedMap.containsKey("eventTime")) {
-            message.eventTime(processTimestamp(renderedMap.get("eventTime")));
-        }
-
-        if (renderedMap.containsKey("deliverAfter")) {
-            message.deliverAfter(processTimestamp(renderedMap.get("deliverAfter")), TimeUnit.MILLISECONDS);
-        }
-
-        if (renderedMap.containsKey("deliverAt")) {
-            message.deliverAt(processTimestamp(renderedMap.get("deliverAt")));
-        }
-
-        if (renderedMap.containsKey("sequenceId")) {
-            message.sequenceId((long) renderedMap.get("sequenceId"));
-        }
-
-        return message.sendAsync();
-    }
-
-    private Long processTimestamp(Object timestamp) {
-        if (timestamp == null) {
-            return null;
-        }
-
-        if (timestamp instanceof Long) {
-            return (Long) timestamp;
-        }
-
-        if (timestamp instanceof ZonedDateTime) {
-            return ((ZonedDateTime) timestamp).toInstant().toEpochMilli();
-        }
-
-        if (timestamp instanceof Instant) {
-            return ((Instant) timestamp).toEpochMilli();
-        }
+    @io.swagger.v3.oas.annotations.media.Schema(
+      title = "JSON strong of the topics schema",
+      description = "Required for connecting with topics using AVRO or JSON schemas"
+    )
+    @PluginProperty
+    private String schemaString;
 
-        if (timestamp instanceof LocalDateTime) {
-            return ((LocalDateTime) timestamp).atZone(ZoneId.systemDefault()).toInstant().toEpochMilli();
-        }
+    @io.swagger.v3.oas.annotations.media.Schema(
+      title = "The topics schema type. If not AVRO or JSON, leave as byte",
+      description = "Required for connecting with topics using AVRO or JSON schemas"
+    )
+    @NotNull
+    @PluginProperty(dynamic = true)
+    @Builder.Default
+    private String schemaType = "byte";
```

Comment:
Better to use an enum here with AVRO and JSON.
And null when there is no schema as byte here means no schema right?

Reply:
yeh essentially. Will make that change